python/lsst/ts/scheduler/__init__.py

```diff
@@ -24,6 +24,19 @@
 except ModuleNotFoundError:
     __version__ = "?"
 
+import logging
+
 from .config_schema import *
 from .scheduler_csc import *
 from .telemetry_stream_handler import *
+
+TRACE_LEVEL_NUM = 5
+logging.addLevelName(TRACE_LEVEL_NUM, "TRACE")
+
+
+def trace(self, message, *args, **kwargs):
+    if self.isEnabledFor(TRACE_LEVEL_NUM):
+        self._log(TRACE_LEVEL_NUM, message, args, **kwargs)
+
+
+logging.getLoggerClass().trace = trace
```

Comment:
Nice! I have been wanting to add TRACE level logging to several CSCs for a while now. Java has it and I miss it in Python.